source/common/upstream/subset_lb.cc

```diff
@@ -330,6 +333,21 @@ bool SubsetLoadBalancer::hostMatches(const SubsetMetadata& kvs, const Host& host
       return false;
     }
 
+    // If configured to do so, we attempt to match the metadata value with the contents
+    // of the list specified on the host.
+    if (list_as_any_) {
+      bool list_match = false;
+      for (const auto& l : entry_it->second.list_value().values()) {
```

Comment:
Is this safe even when the value isn't a list value? (e.g. protobuf produces an empty list when the type of the entry_it->second is a string)